src/events/index.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+
+const ElementHandlers: WeakMap<Element, Map<string, Function>> = new WeakMap();
+
+const returnsFalse = () => false;
+const returnsTrue = () => true;
+const persist = function() {
+  this.isPersistent = returnsTrue;
+};
+const isPropagationStopped = function() {
+  return this.cancelBubble;
+};
+const isDefaultPrevented = function() {
+  return this.defaultPrevented;
+};
+
+const handlerKey = (name, capturing) => `${name}-${String(capturing)}`;
+
+const shouldUseCapture = name =>
+  name === 'focus' ||
+  name === 'blur' ||
+  name === 'scroll' ||
+  name === 'cancel' ||
+  name === 'close';
+
+function handlerProxy(event: Event) {
+  const handlers = ElementHandlers.get(this);
+  if (!handlers) return;
+
+  const handler = handlers.get(
+    handlerKey(event.type, event.eventPhase === event.CAPTURING_PHASE)
+  );
+
+  const syntheticEvent = (event: any);
+  syntheticEvent.persist = persist;
+  syntheticEvent.isPersistent = returnsFalse;
+  syntheticEvent.isDefaultPrevented = isDefaultPrevented;
+  syntheticEvent.isPropagationStopped = isPropagationStopped;
+  syntheticEvent.nativeEvent = event;
```

Comment:
this is nice and easy but i think we'll need a real separate object, from the event vs pretending there are two different ones, unless we don't care that `.type` is likely to be wrong for change events (the native event will be input)